### ISSUE TITLE
[feature]: Create a sample app using SpringBoot and MySQL

### ISSUE THREAD

**Issue by Swpn0neel** [keploy/keploy#2111]
### Is there an existing feature request for this?

- [X] I have searched the existing issues

### Summary

A Node.js application with a MySQL database and GraphQL API for data queries and mutations. It needs to come with proper documentation!

### Why should this be worked on?

There's currently no such sample app with us.

### Repository

samples-java

**Comment by jaiakash** [comment]
Hi @Swpn0neel, in the issue title, you mentioned springboot but in description you mentioned Nodejs. Can you please confirm which one?

**Comment by Swpn0neel** [comment]
> Hi @Swpn0neel, in the issue title, you mentioned springboot but in description you mentioned Nodejs. Can you please confirm which one?

Sorry, my bad... it's would be a Node.js application

**Comment by yaten2302** [comment]
Hey @Swpn0neel sir, I've good knowledge of Nodejs and MySql, may I work on this issue?
For this issue, if I'm not wrong, then we've to create a CRUD app with MySql with graphql api and perform addition, deletion etc on it, am I right?

**Comment by Adity20** [comment]
I would like to collaborate on this one @yaten2302 .

**Comment by yaten2302** [comment]
Yes sure @Adity20 , we can definitely collaborate on this issue 👍 
If I'm not wrong, then this is a CRUD app with graphql. What we can do is that each one of us can perform 2 operations, from these 4 - addition, deletion, update and reading.
What say @Adity20 ?

**Comment by yaten2302** [comment]
But, first lets wait for this issue to get assigned to us, then we can start.

**Comment by Adity20** [comment]
> Yes sure @Adity20 , we can definitely collaborate on this issue 👍 If I'm not wrong, then this is a CRUD app with graphql. What we can do is that each one of us can perform 2 operations, from these 4 - addition, deletion, update and reading. What say @Adity20 ?

Yeah sounds good

**Comment by yaten2302** [comment]
@Adity20 , can we work on my [fork](https://github.com/yaten2302/keploy/), or you would like to create it, so that only 1 PR gets created in the end, I'm comfortable in both ways 👍

**Comment by yaten2302** [comment]
@Adity20 , what I was thinking is that instead of creating HTML files or something, we can use ExpressJs, what say?

**Comment by starvader13** [comment]
> @Adity20 , what I was thinking is that instead of creating HTML files or something, we can use ExpressJs, what say?

Hey, @yaten and @Adity20 I have a pretty good hands-on with the tech-stack mentioned above. I could be a helping hand too. Just in case, do you guys need someone else to collaborate too?

**Comment by Adity20** [comment]
> @Adity20 , what I was thinking is that instead of creating HTML files or something, we can use ExpressJs, what say?

Yeah nice thought

**Comment by Adity20** [comment]
I'll fork your repo then send me the link over slack.

**Comment by yaten2302** [comment]
@Adity20 , here's the [fork](https://github.com/yaten2302/samples-typescript.git). You, can fork this repo and we can start with the work now 👍

**Comment by yaten2302** [comment]
@Adity20 , now ig we're good to go with the work, I've done the basic setup and now we just have to add the 4 cmds. I can do for `get` and `create`? If you want to work on any of these, which I just mentioned, kindly feel free to mention here 👍

**Comment by Adity20** [comment]
Looks good @yaten2302. I'll work on the other two then

**Comment by Adity20** [comment]
@yaten2302 I have added the other two operations kindly review the pr in your fork

**Comment by yaten2302** [comment]
@Adity20 , I've merged your PR, also I wanted to ask that if you've MySql installed on your device, could you also make the requests and use the keploy commands to record the testcases, actually, I don't have it on my device, that's why I was asking. Feel free to LMK, you would like me record the test cases 👍 no issues at all 👍

**Comment by yaten2302** [comment]
@Adity20 , actually, I'm facing some issues with MySql on my Ubuntu, can you run the keploy command on your device and commit that?

**Comment by yaten2302** [comment]
@Swpn0neel sir, we both have created a PR for the sample app, but on my device I'm facing an issue, that when I'm running the `server.js`(the main file), then it's showing an error. The error is related to mysql. Mysql is installed on my ubuntu, but I'm not really sure, why this error is occurring, could you please guide regarding this. <br>
![image](https://github.com/user-attachments/assets/9553c719-9f56-44d8-9934-be659fff726f)

**Comment by Adity20** [comment]
sorry brother @yaten2302 I got busy with some university work. I'll just do the needful right now

**Comment by Adity20** [comment]
I have created another sample app using graphql @yaten2302 give it a look
https://github.com/Adity20/crud_mysql_graphql

Working fine with no errors

**Comment by yaten2302** [comment]
@Adity20 , it's good that you've created another app for this, but ig we should go ahead with the previous one only, because I haven't done anything in your repo, it's all your own. Since, this issue was in collaboration and a level-3 one, I think we should try to fix errors in our previous code only rather than creating a new one separately. 

Also, I don't think that there are any errors in the PR which we've opened, it's just that on my device, there is some issue in setting up MySql and that's why I'm not able to test the app and which is why I asked you to test it on your linux ditro/mac... (since, keploy doesn't works on windows)

**Comment by Adity20** [comment]
Yeah yeah that's not an issue brother I was just trying to make something out of graphql and mysql so did.
And I'll test the app and will commit the dur.
We can add this as another sample app I guess @Swpn0neel

**Comment by Swpn0neel** [comment]
Great, you can create a new issue, and assign it to yourself for that!

**Comment by Adity20** [comment]
Okay @Swpn0neel

**Comment by yaten2302** [comment]
@Swpn0neel @Sonichigo sir, actually, in here, the PR is created and the app is working and well documented, but actually, we're not able to test it using keploy because there's some issue on my device with mysql and @Adity20 is trying to fix the keploy env error (ig on his device, keploy is not opening or something).
May we create a PR for this issue also in the leaderboard or should we wait for the issue to be resolved, so that first the keploy's test case dir is created with all the test cases and then we should create our PR in `.github`?
